### DISCUSSION TITLE
Any reason to use setQueryData in SSR as opposed to prefetchQuery?

### DISCUSSION THREAD

**Discussion by rynslmns** [TanStack/query#6919] [General] [answered]
I have data that I am loading in getServerSideProps and passing to the client. I am currently doing:
`await queryClient.prefetchQuery('user_id', () => ({ user_id: 1 })`.

Is there anything wrong with that? Or should I be doing:
`queryClient.setQueryData('user_id', () => ({ user_id: 1 })`

Curious if there are any performance issues or other things I may be missing?

**Comment by TkDodo** [comment, marked as answer]
they'll both do roughly the same thing
